src/Symfony/Bridge/Doctrine/Form/ChoiceList/DoctrineChoiceLoader.php

```diff
@@ -93,9 +98,7 @@ public function loadChoiceList($value = null)
             ? $this->objectLoader->getEntities()
             : $this->manager->getRepository($this->class)->findAll();
 
-        $this->choiceList = $this->factory->createListFromChoices($objects, $value);
-
-        return $this->choiceList;
+        return $this->choiceList = new ArrayChoiceList($objects, $value);
```

Comment:
The test with the property path added in #17694 is not broken (ref https://github.com/symfony/symfony/pull/17694/files#diff-641188f07a887e51cad48e9d72d69d07R754)